pkg/ddevapp/config.go

```diff
@@ -94,21 +98,43 @@ func (c *Config) Write() error {
 	return nil
 }
 
-// Read app configuration from a specified location on disk.
+// Read app configuration from a specified location on disk, falling back to defaults for config
+// values not defined in the read config file.
 func (c *Config) Read() error {
 	source, err := ioutil.ReadFile(c.ConfigPath)
 	if err != nil {
 		return err
 	}
 
-	err = yaml.Unmarshal(source, c)
+	// Read config values from file into its own empty struct.
+	fromFile := &Config{}
+	err = yaml.Unmarshal(source, fromFile)
+	if err != nil {
+		return err
+	}
+
+	// Reflect non-empty values fromFile into c *Config.
+	fields, err := reflections.Items(fromFile)
 	if err != nil {
 		return err
 	}
 
+	for field, val := range fields {
```

Comment:
If all we're doing is setting defaults on a few fields, series of if statements checking for empty values and then setting them as appropriate seems preferable to bringing reflection into this, especially for such a limited use case?